src/Content/PostActions.vala

```diff
@@ -52,19 +52,52 @@ public class PostActions : Gtk.Widget {
     set {
       displayed_post = value;
 
-      // Set the information on the UI
-      likes_counter.label   = displayed_post != null ? DisplayUtils.shortened_metric (displayed_post.liked_count)    : "(null)";
-      reposts_counter.label = displayed_post != null ? DisplayUtils.shortened_metric (displayed_post.reposted_count) : "(null)";
-      replies_counter.label = displayed_post != null ? DisplayUtils.shortened_metric (displayed_post.replied_count)  : "(null)";
-
-      // Set up options menu
       if (displayed_post != null) {
+        likes_counter.label   = DisplayUtils.shortened_metric (displayed_post.liked_count);
+        likes_button.sensitive = true;
+        reposts_counter.label = DisplayUtils.shortened_metric (displayed_post.reposted_count);
+        reposts_button.sensitive = true;
+        replies_counter.label = DisplayUtils.shortened_metric (displayed_post.replied_count);
+        replies_button.sensitive = true;
+
+        debug("Likes: %d; Label: %s; Liked: %s", displayed_post.liked_count, likes_counter.label, displayed_post.is_favourited ? "yes" : "no");
+
+        if (displayed_post.is_favourited) {
+          likes_counter.icon_name = "liked-symbolic";
+          likes_button.add_css_class ("liked");
```

Comment:
Please use the `conditional_css` method from the `DisplayUtils`.

Reply:
Ah, yes, I'd seen that class and didn't think about it because it seemed like a quick and easy thing to write.

Reply:
At the end, its just the same lines of code a lot of time, so I thought it makes sense to de-deduplicate that. It's also makes the code more readable IMHO.

Reply:
It makes sense to ensure the pattern is consistently applied. I've got the add and remove here. But it's easy to put the add in and forget the remove. Especially when conditions get more complex.